### PR TITLE
[C2y] Add test coverage for WG14 N3370

### PR DESCRIPTION
**Pull request by AaronBallman**
This paper added case ranges in switch statements, which is a GNU extension Clang has supported since at least Clang 3.0.

It updates the diagnostics to no longer call this a GNU extension except in C++ mode.

### CODE REVIEW

File: clang/test/C/C2y/n3370.c [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  // there is no duplicate value 20 to begin with.
+  case 10 ... 20: break; /* expected-error {{duplicate case value '11'}}
+                            expected-note {{previous case defined here}}
+                            gnu-warning {{case ranges are a GNU extension}}
+                            ped-warning {{case ranges are a C2y extension}}
+                            pre-warning {{case ranges are incompatible with C standards before C2y}}
+                          */
+  case 11: break;        /* expected-note {{previous case defined here}}
+                          */
+  case 11 ... 14: break; /* expected-error {{duplicate case value '20'}}
```
Comment (erichkeane) [inline clang/test/C/C2y/n3370.c:80]: Oh my :)

**Review by erichkeane — COMMENTED** [review]
Release note to say we enable this in C2Y?

**Comment by AaronBallman** [comment]
> Release note to say we enable this in C2Y?

I'm not opposed, but we've always supported this feature. Similar to `i` and `j` suffixes, I didn't add a release for imaginary constants which are in the same boat (extensions we've supported forever but are now C2y features). WDYT?

**Comment by erichkeane** [comment]
> > Release note to say we enable this in C2Y?
> 
> I'm not opposed, but we've always supported this feature. Similar to `i` and `j` suffixes, I didn't add a release for imaginary constants which are in the same boat (extensions we've supported forever but are now C2y features). WDYT?

There is a minor value (at least for messaging!) for a release note that says "we looked at implementing N3370, saw it was implemented, so added sufficient testing to validate it, so now you can use it without getting GNU-ext warnings in C"

**Review by Sirraide — APPROVED** [review]
Changes lgtm.

I don’t really have a strong opinion on the release note: on the one hand,  writing one for this does feel a bit like we’d be writing a release note for C2y itself rather than for Clang, but on the other, I guess adding something like ‘Clang now diagnoses case ranges as a Clang extension rather than a GNU extension’ would be an option. Iirc I added a similar release note when I changed some variadic macro warnings.

**Comment by AaronBallman** [comment]
I've fixed up a failing test caught by precommit CI and added a release note.

**Pull request merged**